snapcraft/_store.py

```diff
@@ -59,25 +54,42 @@ def logout():
 
 
 def upload(snap_filename):
-    if not os.path.exists(snap_filename):
-        raise FileNotFoundError(snap_filename)
-    else:
-        snap_name = _get_name_from_snap_file(snap_filename)
-        logger.info('Uploading existing {}.'.format(snap_filename))
+    logger.info('Uploading existing {}.'.format(snap_filename))
 
-        storeapi.upload(snap_filename, snap_name)
+    try:
+        store = storeapi.StoreClient()
+        result = store.upload(snap_filename)
+    except storeapi.InvalidCredentialsError:
+        logger.info('No valid credentials found.'
+                    ' Have you run "snapcraft login"?')
+        return
 
+    success = result.get('success', False)
+    errors = result.get('errors', [])
+    app_url = result.get('application_url', '')
+    revision = result.get('revision')
 
-def _get_name_from_snap_file(snap_path):
-    with tempfile.TemporaryDirectory() as temp_dir:
-        subprocess.check_call(
-            ['unsquashfs', '-d', os.path.join(temp_dir, 'squashfs-root'),
-             snap_path, '-e', os.path.join('meta', 'snap.yaml')])
-        with open(os.path.join(
-                temp_dir, 'squashfs-root', 'meta', 'snap.yaml')) as yaml_file:
-            snap_yaml = yaml.load(yaml_file)
+    # Print another newline to make sure the user sees the final result of the
+    # upload (success/failure).
+    logger.info('')
```

Comment:
This should be a `print()`, it makes little sense to log an empty line if it is for ui purposes